source/js/components/post-code-validation.js

```diff
@@ -0,0 +1,58 @@
+// Importing JSON list of countries and post-code info.
+// This info is also shared with donate/forms.py to check on the backend.
+try {
+  var countriesAndPostCodes = require("./post-codes-list");
+} catch (err) {
+  console.error(
+    "Could not import post-code data. Zip code is required by default."
+  );
+}
+// The container for the post-code input field.
+const postCodeInput = document.querySelector(".post-code-input");
+
+// The container that houses both the post code and city inputs.
+const formAndCityContainer = document.querySelector(".form__group--city-post");
+
+// The select object that allows users to select their country.
+const countrySelector = document.getElementById("id_country");
+
+// If you are on the form page, check if users country uses post code.
+if (countrySelector) {
+  checkForCountryPostCode();
+}
+// Event listener for checking post-code when user selects new country.
+function countryPostCodeValidation() {
+  if (countrySelector && countriesAndPostCodes) {
+    countrySelector.addEventListener("change", (e) => {
```

Comment:
If we're not using the event, no need to explicitly capture it as call argument.
```suggestion
    countrySelector.addEventListener("change", () => {
```